src/Notifier.cpp

```diff
@@ -138,15 +149,68 @@ void Notifier::sendDevicePresenceNotification(DevicePresenceInfo& info)
     Rule rule = Rule::fromString(info.device_rule);
 
     std::ostringstream body;
-    body << event_str << ' ' << rule.getName() << ": " << target_str;
+    if (info.event == DeviceManager::EventType::Remove) {
+        body << rule.getName();
+    } else {
+        body << rule.getName() << ": " << target_str << "ed";
+    }
 
-    notify::Notification n("USBGuard", body.str());
+    notify::Notification n(event_str + " - USBGuard", body.str());
     n.setTimeout(5000);
     n.setCategory("device");
+
+    if (info.event == DeviceManager::EventType::Insert) {
+        _actionsCallbackUserData.info = &info;
+        if (info.target != Rule::Target::Allow) {
+            n.addAction("allow", "Allow",
+                &Notifier::actionsCallbackWrapper,
+                &_actionsCallbackUserData);
+        }
+        n.addAction("reject", "Reject",
+            &Notifier::actionsCallbackWrapper,
+            &_actionsCallbackUserData);
+    }
+
     if (!n.show()) {
         throw std::runtime_error("Failed to show notification");
     }
     // TODO serialize
 }
 
+void Notifier::actionsCallback(std::string action_id, DevicePresenceInfo* info)
+{
+    using namespace usbguard;
+
+    try {
+        if (action_id == "allow") {
```

Comment:
If I remember correctly, the usbguard API does provide a targetFromString method, which can convert a target string into ```Rule::Target```, see https://usbguard.github.io/documentation/doc-0-7-6/classusbguard_1_1Rule.html#a29ef58acc9cf41bc9a0504cdd7663d2c
Could you try it out?

Reply:
Thanks for taking the time to look at the pull request source code.

I've followed your recommendation and changed the lines in Notifier::actionsCallback, as well as those in Notifier::sendDevicePresenceNotification.

I've also noticed this new feature was throwing a segmentation fault when restarting usbguard service, so I've also fixed it.